module/data/item/discipline.mjs

```diff
@@ -1,41 +1,341 @@
+      nextTalentLpCost: ED4E.legendPointsCost[ nextLevel + ED4E.lpIndexModForTier[ this.currentTier ] ],
+      talentRequirements,
+    };
+  }
+
+  /** @inheritDoc */
+  get increaseValidationData() {
+    if ( !this.isActorEmbedded ) return undefined;
+
+    const { learn, nextTalentLpCost, talentRequirements } = this.increaseData;
+    const validationData = {
+      [ED4E.validationCategories.resources]:               [
+        {
+          name:      "ED.Dialogs.Legend.Validation.availableLp",
+          value:     this.requiredLpForIncrease,
+          fulfilled: this.requiredLpForIncrease <= this.parentActor.currentLp,
+        },
+        {
+          name:      "ED.Dialogs.Legend.Validation.availableMoney",
+          value:     this.requiredMoneyForIncrease,
+          fulfilled: this.requiredMoneyForIncrease <= this.parent.actor.currentSilver,
+        },
+      ],
+      [ED4E.validationCategories.talentsRequirement]: talentRequirements,
+    };
+    if ( !learn ) validationData[ED4E.validationCategories.newAbilityLp] = [
+      {
+        name:      "ED.Dialogs.Legend.Validation.talentOptionLp",
+        value:     nextTalentLpCost,
+        fulfilled: nextTalentLpCost <= this.parentActor.currentLp,
+      },
+    ];
+    return validationData;
+  }
+
+  /** @inheritDoc */
+  get learnRules() {
+    return game.i18n.localize( "ED.Dialogs.Legend.Rules.disciplineLearnShortRequirements" );
+  }
+
+  /** @inheritDoc */
+  get requiredMoneyForIncrease() {
+    return ED4E.disciplineTeacherCost[ this.level + 1 ];
+  }
+
+  /**
+   * Get all talents from the discipline associated with this item.
+   * @type {ItemEd[]}
+   */
+  get talentsFromDiscipline() {
+    return this.parentActor.itemTypes.talent.filter(
+      talent => talent.system.source.class === this.parent.uuid
+    );
+  }
+
+  get _talentRequirementsStandard() {
+    const nextLevel = this.level + 1;
+    const disciplineTalents = this.getTalentsByCategory( "discipline" );
+    const unfulfilledTalents = disciplineTalents.filter( talent => talent.system.level < nextLevel );
+    const fulfilled = unfulfilledTalents.length === 0;
+    const requirementValue = game.i18n.format(
+      `ED.Dialogs.Legend.Validation.${
+        fulfilled
+          ? "talentsRequirementFailed"
+          : "talentsRequirementFulfilled"
+      }`,
+      {
+        requiredLevel:      nextLevel,
+        unfulfilledTalents: unfulfilledTalents.map(
+          talent => linkForUuidSync( talent.uuid )
+        ).join( "\n" ),
+      }
+    );
+    return [
+      {
+        name:      ED4E.circleTalentRequirements[
+          game.settings.get( "ed4e", "lpTrackingCircleTalentRequirements" )
+        ],
+        value:     requirementValue,
+        fulfilled,
+      }
+    ];
+  }
+
+  get _talentRequirementsOptional() {
+    const nextLevel = this.level + 1;
+    const allCorrespondingTalents = this.talentsFromDiscipline;
+    const hasTalentFromCurrentCircle = allCorrespondingTalents.some(
```

Comment:
I understand this to only check if there is at least one talent at that level, but the idea is from the table, that there are 4+circle (or 5?!?!) talents at that level